src/schnorr.rs

```diff
@@ -12,55 +13,75 @@ use crate::{
 pub struct ID {
-    /// The ID
+    /// ID
     pub id: Scalar,
-    /// The public schnorr response
-    pub kG: Point,
-    /// The aggregate of the schnorr committed values
-    pub kca: Scalar,
+    /// Commitment to the proof random value
+    pub random_commitment: Point,
+    /// Sigma protocol response
+    pub sigma_response: Scalar,
 }
 
 #[allow(non_snake_case)]
 impl ID {
-    /// Construct a new schnorr ID which binds the passed `Scalar` `id` and `Scalar` `a`, with a
-    /// zero-knowledge proof of ownership of `a`.  The `ctx` is a common reference string used to
+    /// Construct a new schnorr ID which binds the passed `Scalar` `id` and `Scalar` `x`, with a
+    /// zero-knowledge proof of ownership of `x`.  The `ctx` is a common reference string used to
     /// prevent replay attacks; it can be any length, but will typically be a `u64` value in
     /// big endian format.
     pub fn new<RNG: RngCore + CryptoRng>(
         id: &Scalar,
-        a: &Scalar,
+        x: &Scalar,
         ctx: &[u8],
         rng: &mut RNG,
     ) -> Self {
-        let k = Scalar::random(rng);
-        let c = Self::challenge(id, &(&k * &G), &(a * &G), ctx);
+        let r = Scalar::random(rng);
+        let random_commitment = r * G;
+        let public_key = x * G;
+        let c = Self::challenge(id, &random_commitment, &public_key, ctx);
+        let sigma_response = r + c * x;
 
         Self {
             id: *id,
-            kG: &k * G,
-            kca: &k + c * a,
+            random_commitment,
+            sigma_response,
         }
     }
 
     /// Compute the schnorr challenge
-    pub fn challenge(id: &Scalar, K: &Point, A: &Point, ctx: &[u8]) -> Scalar {
+    pub fn challenge(
+        id: &Scalar,
+        random_commitment: &Point,
+        public_key: &Point,
+        ctx: &[u8],
+    ) -> Scalar {
         let mut hasher = Sha256::new();
         let tag = "WSTS/polynomial-constant";
 
         hasher.update(tag.as_bytes());
         hasher.update(id.to_bytes());
-        hasher.update(K.compress().as_bytes());
-        hasher.update(A.compress().as_bytes());
+
+        hasher.update(random_commitment.compress().as_bytes());
+        hasher.update(public_key.compress().as_bytes());
         hasher.update(ctx);
 
         hash_to_scalar(&mut hasher)
     }
 
     /// Verify the proof
-    pub fn verify(&self, A: &Point, ctx: &[u8]) -> bool {
-        let c = Self::challenge(&self.id, &self.kG, A, ctx);
-        &self.kca * &G == &self.kG + c * A
+    pub fn verify(&self, public_key: &Point, ctx: &[u8]) -> bool {
+        let c = Self::challenge(&self.id, &self.random_commitment, public_key, ctx);
+        &self.sigma_response * &G == &self.random_commitment + c * public_key
+    }
+
+    /// Zero out the schnorr proof
+    pub fn zero(&mut self) {
+        self.random_commitment = Point::new();
+        self.sigma_response = Scalar::zero();
+    }
```

Comment:
This function should probably be named `zeroize`, since `zero` suggests that it will return `zero`.

Reply:
Same thing for `PolyCommitment::zero`.

Reply:
Done